packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -447,6 +449,31 @@ export class ConstructNode {
     return Array.from(ret);
   }
 
+  /**
+   * Add an ordering dependency on another Construct.
+   *
+   * All constructs in the dependency's scope will be deployed before any
+   * construct in this construct's scope.
+   */
+  public addDependency(...dependencies: IConstruct[]) {
+    for (const dependency of dependencies) {
+      this.dependencies.add(dependency);
+    }
+  }
+
+  /**
+   * Return all dependencies registered on this node and its ancestors.
+   */
+  public myDependencies(): IConstruct[] {
+    const ret = new Set<IConstruct>();
+    let node: ConstructNode | undefined = this;
+    while (node !== undefined) {
+      addSet(ret, node.dependencies);
```

Comment:
I think it would have been easier to read if you had just in-lined the loop here.